src/main/java/com/squareup/javapoet/ClassName.java

```diff
@@ -87,6 +87,18 @@ public ClassName nestedClassNamed(String name) {
     return names.subList(1, names.size());
   }
 
+  /**
+   * Returns a class that shares the same enclosing package or class. If this class is enclosed by
+   * another class, this is equivalent to {@code enclosingClassName().nestedClassNamed(name)}.
+   * Otherwise it is equivalent to {@code get(packageName(), name)}.
+   */
+  public ClassName peerClass(String name) {
```

Comment:
I used `nestedClassNamed` for nesting. Either append 'Named' to this method or remove it from mine.